src-java/swmanager-topology/swmanager-storm-topology/src/main/java/org/openkilda/wfm/topology/switchmanager/service/SwitchRuleService.java

```diff
@@ -325,6 +331,89 @@ private Predicate<SpeakerData> buildPredicate(InstallRulesAction action) {
         }
     }
 
+    private boolean isInstallRequired(DeleteRulesAction action) {
+        return action == DROP_ALL_ADD_DEFAULTS || action == REMOVE_ADD_DEFAULTS || action == OVERWRITE_DEFAULTS;
+    }
+
+    private Predicate<SpeakerData> buildDeletePredicate(DeleteRulesAction action) {
```

Comment:
Does it make sense to extract this code into a dedicated builder class?

Reply:
Moved to separate class.